core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortParser.java

```diff
@@ -57,6 +58,8 @@
  */
 public class GeoDistanceSortParser implements SortParser {
 
+    private static final ParseField REVERSE_FIELD = new ParseField("reverse").withAllDeprecated("order");
```

Comment:
I'm not sure this works as expected, we are not replacing "reverse" with "order", aren't we? That would mean we need to accept `"order" : true` now, which I think we don't. I'd put some note here like "replaced by order" like we do in the RangeQueryParse here.